ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -1773,9 +1773,14 @@ public Table getTable(final String dbName, final String tableName, String metaTa
    */
   private ValidWriteIdList getValidWriteIdList(String dbName, String tableName) throws LockException {
     ValidWriteIdList validWriteIdList = null;
-    long txnId = SessionState.get() != null && SessionState.get().getTxnMgr() != null ? SessionState.get().getTxnMgr().getCurrentTxnId() : 0;
+    SessionState sessionState = SessionState.get();
+    HiveTxnManager txnMgr = sessionState != null? sessionState.getTxnMgr() : null;
```

Comment:
Can you hack a unit test, to set the session state as null and then fetch get_partitions_by_names()?